fuel-relayer/src/db.rs

```diff
@@ -0,0 +1,124 @@
+use async_trait::async_trait;
+use fuel_core_interfaces::{
+    common::fuel_storage::{
+        StorageAsMut,
+        StorageMutate,
+    },
+    model::{
+        BlockHeight,
+        CheckedMessage,
+        DaBlockHeight,
+        SealedFuelBlock,
+    },
+};
+use fuel_database::{
+    tables::Messages,
+    Column,
+    Database,
+    KvStoreError,
+};
+use std::sync::Arc;
+
+pub(crate) const FINALIZED_DA_HEIGHT_KEY: &[u8] = b"finalized_da_height";
+pub(crate) const LAST_PUBLISHED_BLOCK_HEIGHT_KEY: &[u8] = b"last_publish_block_height";
+
+/// Manages state related to supported external chains.
+#[async_trait]
+pub trait RelayerDb: StorageMutate<Messages, Error = KvStoreError> + Send + Sync {
+    /// Add bridge message to database. Messages are not revertible.
+    async fn insert_message(&mut self, message: &CheckedMessage);
+
+    /// current best block number
+    async fn get_chain_height(&self) -> BlockHeight;
+
+    /// Returns the sealed block at the `height`.
+    async fn get_sealed_block(&self, height: BlockHeight)
+        -> Option<Arc<SealedFuelBlock>>;
+
+    /// set finalized da height that represent last block from da layer that got finalized.
+    async fn set_finalized_da_height(&self, block: DaBlockHeight);
+
+    /// Assume it is always set as initialization of database.
+    async fn get_finalized_da_height(&self) -> Option<DaBlockHeight>;
+
+    /// Get the last fuel block height that was published to the da layer.
+    async fn get_last_published_fuel_height(&self) -> Option<BlockHeight>;
+
+    /// Set the last fuel block height that was published to the da layer.
+    async fn set_last_published_fuel_height(&self, block_height: BlockHeight);
+}
+
+#[async_trait::async_trait]
+impl RelayerDb for Database {
+    async fn insert_message(&mut self, message: &CheckedMessage) {
+        let _ = self
+            .storage::<Messages>()
+            .insert(message.id(), message.as_ref());
```

Comment:
Doesn't a result need to be handled here?

Reply:
That code was before=)
<img width="1132" alt="image" src="https://user-images.githubusercontent.com/18346821/205543202-6c162bfe-421e-4d5b-a4d8-056863ed9aa9.png">


Of course we need to handle it and return a result, but it is not part of this PR=)

Reply:
Didn't you say you were going to fix this already? 😆 

https://github.com/FuelLabs/fuel-core/issues/563#issuecomment-1216085668